### ISSUE TITLE
[Proposal] support section aliases / alias propagation

### ISSUE THREAD

**Issue by ES-Alexander** [Keats/gutenberg#2112]
## Idea
Setting `aliases` on a page is really useful, but it would be even more useful if they automatically propagated to any children of that page.

## Example(s)
I'm thinking something like 

`mydocs/v1.5.0/...` aliased to be accessible via `mydocs/latest/...`, which then automatically applies to all the pages within that version (section). That then avoids needing to add a slightly modified alias to every page within that section, and recursively to any sub-sections, and also massively simplifies the required maintenance when new versions are released that take over the aliases like `latest`/`beta`/ `stable`.

## Additional Information
I was considering proposing this as a similar but separate feature to co-exist with the present aliases, but I can't think of any examples where it would be desirable to _not_ have this kind of propagation, so maybe it can just be an improvement to the aliasing behaviour.

**Comment by Keats** [comment]
Aliases redirect to the current page/section, so `mydocs/latest/..` will redirect to `mydocs/v1.5.0/`. If you plan to have the URL stay like in https://docs.rs/tera/latest/tera/# that would need to be a different feature.

**Comment by ES-Alexander** [comment]
The idea is not to have the URL stay, the point is to have child pages _also_ have the same (base) redirection, without it needing to be explicitly specified for each child. To clarify,

If I alias the `mydocs/v1.5.0/` section to `mydocs/latest/` then I would like it for all of its subpages (e.g. `mydocs/v1.5.0/intro/`, `mydocs/v1.5.0/quick-start/`, `mydocs/v1.5.0/features/`, etc) to automatically have the `latest` base alias (e.g. `mydocs/latest/intro` should redirect to `/mydocs/v1.5.0/intro` _because of the section alias_).

I'm imagining this would be applied by recursively adding the alias of any section as a base alias to all of its children, extended by the (relative) path from the section page to the relevant child page.

Does that make my intent clearer? :-)

**Comment by Keats** [comment]
Yep I see, I was just wondering if you wanted persistent URLs like docs.rs.
I _think_ it makes sense?

**Comment by ES-Alexander** [comment]
> I was just wondering if you wanted persistent URLs like docs.rs.

That seems like a potentially useful feature, but it's not the one I'm asking for / suggesting in this proposal :-)
